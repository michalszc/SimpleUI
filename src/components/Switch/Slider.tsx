import React, { FC } from "react";
import styled, { css } from "styled-components";
import { StyleProps } from "../../utils";
import Styles from "../../utils/styles";

export interface SliderProps extends StyleProps {
    isDisabled: boolean;
    shape: 'circle' | 'square' | 'roundedSquare';
}

const SliderComponent = styled.span<SliderProps>`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: ${ ({ theme }) =>  theme.colors.grey[500] };
    border-radius: ${ ({ shape, theme }) => {
        switch (shape) {
            case 'circle':
                return '34px';
            case 'square':
                return '0';
            case 'roundedSquare':
                return theme.sizes[1.5];
        }
    }};
    -webkit-transition: .4s;
    transition: .4s;

    &:before {
        position: absolute;
        content: "";
        height: 26px;
        width: 26px;
        left: 4px;
        bottom: 4px;
        background-color: white;
        -webkit-transition: .4s;
        transition: .4s;
        border-radius: ${ ({ shape, theme }) => {
            switch (shape) {
                case 'circle':
                    return '50%';
                case 'square':
                    return '0';
                case 'roundedSquare':
                    return theme.sizes[1];
            }
        }};
    }

    ${({ isDisabled, theme }) => isDisabled && css`
        opacity: 0.45;
        cursor: not-allowed;
    `}
`;  

// !important;

const Slider: FC<SliderProps> = ({
    shape, isDisabled, ...props
}) => (
    <Styles>
        <SliderComponent shape={shape} isDisabled={isDisabled} className={"simpleui-switch-slider"}/>
    </Styles>
);

export default Slider;