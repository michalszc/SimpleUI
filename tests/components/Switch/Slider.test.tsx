import React from "react";
import { render } from "@testing-library/react";
import Slider from "../../../src/components/Switch/Slider";

describe("Slider", () => {
    test("should properly render Slider component - shape circle, enabled", () => {
        const container = render(<Slider shape={'circle'} isDisabled={false} isReadonly={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Slider component - shape square, enabled", () => {
        const container = render(<Slider shape={'square'} isDisabled={false} isReadonly={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Slider component - shape roundedSquare, enabled", () => {
        const container = render(<Slider shape={'roundedSquare'} isDisabled={false} isReadonly={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Slider component - shape circle, disabled", () => {
        const container = render(<Slider shape={'circle'} isDisabled={true} isReadonly={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Slider component - shape square, disabled", () => {
        const container = render(<Slider shape={'square'} isDisabled={true} isReadonly={false}/>);
        expect(container).toMatchSnapshot();
    });
    test("should properly render Slider component - shape roundedSquare, disabled", () => {
        const container = render(<Slider shape={'roundedSquare'} isDisabled={true} isReadonly={false}/>);
        expect(container).toMatchSnapshot();
    });
});
