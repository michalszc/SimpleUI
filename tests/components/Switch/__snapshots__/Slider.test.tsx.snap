// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should properly render Slider component - shape circle, disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXHqe hsAQcy simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      >
        <span
          class="sc-gswNZR drzfKv simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXHqe hsAQcy simpleui-switch-slider"
      id="4fzzzxjylrx"
      mode="light"
      shape="circle"
    >
      <span
        class="sc-gswNZR drzfKv simpleui-switch-dot"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slider should properly render Slider component - shape circle, enabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXHqe gYijzI simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      >
        <span
          class="sc-gswNZR drzfKv simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="circle"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXHqe gYijzI simpleui-switch-slider"
      id="4fzzzxjylrx"
      mode="light"
      shape="circle"
    >
      <span
        class="sc-gswNZR drzfKv simpleui-switch-dot"
        id="4fzzzxjylrx"
        mode="light"
        shape="circle"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slider should properly render Slider component - shape roundedSquare, disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXHqe dMBhPM simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
      >
        <span
          class="sc-gswNZR dxaNzs simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="roundedSquare"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXHqe dMBhPM simpleui-switch-slider"
      id="4fzzzxjylrx"
      mode="light"
      shape="roundedSquare"
    >
      <span
        class="sc-gswNZR dxaNzs simpleui-switch-dot"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slider should properly render Slider component - shape roundedSquare, enabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXHqe dmLylq simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
      >
        <span
          class="sc-gswNZR dxaNzs simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="roundedSquare"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXHqe dmLylq simpleui-switch-slider"
      id="4fzzzxjylrx"
      mode="light"
      shape="roundedSquare"
    >
      <span
        class="sc-gswNZR dxaNzs simpleui-switch-dot"
        id="4fzzzxjylrx"
        mode="light"
        shape="roundedSquare"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slider should properly render Slider component - shape square, disabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXHqe JYBJd simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
      >
        <span
          class="sc-gswNZR WwCXX simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="square"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXHqe JYBJd simpleui-switch-slider"
      id="4fzzzxjylrx"
      mode="light"
      shape="square"
    >
      <span
        class="sc-gswNZR WwCXX simpleui-switch-dot"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Slider should properly render Slider component - shape square, enabled 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXHqe dlnOhD simpleui-switch-slider"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
      >
        <span
          class="sc-gswNZR WwCXX simpleui-switch-dot"
          id="4fzzzxjylrx"
          mode="light"
          shape="square"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXHqe dlnOhD simpleui-switch-slider"
      id="4fzzzxjylrx"
      mode="light"
      shape="square"
    >
      <span
        class="sc-gswNZR WwCXX simpleui-switch-dot"
        id="4fzzzxjylrx"
        mode="light"
        shape="square"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
